### PR TITLE
Edcsa sign with random k

### PR DESCRIPTION
**Pull request by akonior**

### CODE REVIEW

File: zkmarek/crypto/ecdsa.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+from dataclasses import dataclass
+
+
+@dataclass(frozen=True)
+class ECDSASignature:
+    r: int
+    s: int
+
+    def to_der(self) -> bytes:
```
Comment (akonior) [inline zkmarek/crypto/ecdsa.py:9]: Add lint to DER spec/format

File: zkmarek/test/test_standard.py [hunk 2]
```diff
@@ -100,6 +116,23 @@ def test_verify_signature(self):
             with self.subTest(f'Test verify method properly process {message} signature'):
                 self.assertEqual(result, standard.verify(z, r_value, s, public_key))
 
+    def test_verify_with_signature_generated_by_cryptography_library(self):
+        private_key = ec.generate_private_key(ec.SECP256K1())
```
Comment (akonior) [inline zkmarek/test/test_standard.py:120]: x1000?

File: zkmarek/crypto/ecdsa.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+        def encode_int(n: int) -> bytes:
+            hex_n = hex(n)[2:]
+            if len(hex_n) % 2 == 1:
+                hex_n = '0' + hex_n
```
Comment (marekkirejczyk) [inline zkmarek/crypto/ecdsa.py:14]: `hex_even_length()`

File: zkmarek/crypto/standard.py [hunk 4]
```diff
@@ -35,6 +37,12 @@ def sign(self, secret_key: int, msg_hash: int, k: int):
         s = (pow(k, -1, n) * (msg_hash + (r * int(hex(secret_key), 16)))) % n
         return r, s
 
+    def sign_with_random_k(self, secret_key: int, msg: bytes) -> ECDSASignature:
```
Comment (marekkirejczyk) [inline zkmarek/crypto/standard.py:40]: Move to `ECDSA.sign`

**Review by marekkirejczyk — APPROVED** [review]

**Pull request merged**